test/pattern/if_case.stmt

```diff
@@ -0,0 +1,7 @@
+40 columns                              |
```

Comment:
I'd move this test to "statement/if_case.stmt" since it's about testing the if-case itself (I think) and not the patterns within it.

Also, can you add another test file ("expression/collection_if_case.stmt") for if-case elements?

```dart
var list = [
  if (obj case true) element
];
```

(And then eventually that will expand to the various ways that you can split inside if-case and patterns inside if-case elements.)

Reply:
For the part on moving it to `statement/if_case.stmt`, that sounds good. Done.

For the second part on if-case elements. This PR doesn't implement if-case elements yet. I'll do that once I implement it next PR.